financial-connections/src/main/java/com/stripe/android/financialconnections/features/networkinglinksignup/LinkSignupHandler.kt

```diff
@@ -0,0 +1,125 @@
+package com.stripe.android.financialconnections.features.networkinglinksignup
+
+import com.stripe.android.core.Logger
+import com.stripe.android.financialconnections.analytics.FinancialConnectionsAnalyticsTracker
+import com.stripe.android.financialconnections.analytics.logError
+import com.stripe.android.financialconnections.model.FinancialConnectionsSessionManifest.Pane
+import com.stripe.android.financialconnections.model.FinancialConnectionsSessionManifest.Pane.LINK_ACCOUNT_PICKER
+import com.stripe.android.financialconnections.model.FinancialConnectionsSessionManifest.Pane.LINK_LOGIN
+import com.stripe.android.financialconnections.model.FinancialConnectionsSessionManifest.Pane.NETWORKING_LINK_SIGNUP_PANE
+import com.stripe.android.financialconnections.navigation.Destination.NetworkingLinkVerification
+import com.stripe.android.financialconnections.navigation.Destination.NetworkingSaveToLinkVerification
+import com.stripe.android.financialconnections.navigation.Destination.Success
+import com.stripe.android.financialconnections.navigation.NavigationManager
+import javax.inject.Inject
+import javax.inject.Provider
+
+internal class LinkSignupHandlerFactory @Inject constructor(
+    private val performLinkSignupForNetworking: Provider<PerformLinkSignupForNetworking>,
+    private val performLinkSignupForInstantDebits: Provider<PerformLinkSignupForInstantDebits>,
+    private val navigationManager: NavigationManager,
+    private val eventTracker: FinancialConnectionsAnalyticsTracker,
+    private val logger: Logger,
+) {
+
+    fun create(isInstantDebits: Boolean): LinkSignupHandler {
+        return if (isInstantDebits) {
+            LinkSignupHandlerForInstantDebits(
+                performLinkSignup = performLinkSignupForInstantDebits.get(),
```

Comment:
maybe we can merge `performLinkSignupForN` with `linkSignupHandlerForN` to avoid layering too much? Found it a bit hard to follow despite the great naming 😅 

Unless we're envisioning reusing `performLinkSignupForN` on other places.